src/rpc/rpc.cc

```diff
@@ -298,7 +298,7 @@ namespace rpc {
               }
 
               p->uncancellable();
-              return send_entry(*p).then_wrapped([this, p = std::move(p)] (auto f) mutable {
+              return futurize_invoke([this, &pp = *p] { return send_entry(pp); }).then_wrapped([this, p = std::move(p)] (auto f) mutable {
```

Comment:
instead of fixing the caller of `connection::send_entry()`, does it make sense to fix `connection::send_entry()` ? as it would reduce the chance of such kind of errors. coroutines should always return a failed future rather than throwing an actual exception.

Reply:
+1 It's better to make send_entry() be noexcept (and maybe even mark it as such)

Reply:
Makes sense, I will move the fix to the callee.

Reply:
Done.